From 92db24adfa941003db1d885df01157056617f30b Mon Sep 17 00:00:00 2001
From: Maks Orlovich <maksim@kde.org>
Date: Sun, 8 May 2011 14:39:03 -0400
Subject: [PATCH] Fix the job-on-hold reuse logic, which caused the
 double-POST problem)

adawit, could you please at least READ what you're backporting if you are
going to be this aggressive? Or better yet, please don't backport anything
that's not fixing a critical bug or is trivial, as per:
http://techbase.kde.org/Policies/Minor_Point_Release_Policy

CCMAIL: adawit@kde.org
BUG: 272466
---
 kio/kio/scheduler.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/kio/kio/scheduler.cpp b/kio/kio/scheduler.cpp
index 55da053..9f5607e 100644
--- a/kio/kio/scheduler.cpp
+++ b/kio/kio/scheduler.cpp
@@ -1151,7 +1151,7 @@ Slave *SchedulerPrivate::heldSlaveForJob(SimpleJob *job)
         bool canJobReuse = (cmd == CMD_GET || cmd == CMD_MULTI_GET);
 
         if (KIO::TransferJob *tJob = qobject_cast<KIO::TransferJob *>(job)) {
-            canJobReuse = cmd == (canJobReuse || cmd == CMD_SPECIAL);
+            canJobReuse = (canJobReuse || cmd == CMD_SPECIAL);
             if (canJobReuse) {
                 KIO::MetaData outgoing = tJob->outgoingMetaData();
                 const QString resume = outgoing.value("resume");
-- 
1.7.4.4

